elements/lisk-p2p/src/validation.ts

```diff
@@ -115,6 +132,23 @@ export const validatePeerInfoList = (
 	}
 };
 
+export const validateBasicPeersInfoList = (
+	rawBasicPeerInfoList: unknown,
+): ReadonlyArray<P2PPeerInfo> => {
+	if (!rawBasicPeerInfoList) {
+		throw new InvalidRPCResponseError('Invalid response type');
+	}
+	const { peers } = rawBasicPeerInfoList as RPCPeerListResponse;
+
+	if (Array.isArray(peers)) {
+		const peerList = peers.map<P2PPeerInfo>(validateBasicPeerInfo);
+
+		return peerList;
+	} else {
```

Comment:
Prefer avoiding `else` by checking `!Array.isArray(peers))`

Reply:
I agree that it's better but if we do this we should also update `validatePeerInfoList` because it also has a similar else condition. Maybe we can clean this up later as part of a different PR?

Reply:
@mitsujutsu I agree with your comment but in this case, the tests fail when we follow your approach.